optimade/server/config.py

```diff
@@ -11,12 +11,12 @@ class NoFallback(Exception):
 class Config:
     """Base class for loading config files and its parameters"""
 
-    index_links_path: Path = Path("./optimade/server/index_links.json")
-    _path: Path = Path("./optimade/server/config.ini")
+    index_links_path: Path = Path(__file__).parent.joinpath("index_links.json")
+    _path: Path = Path(__file__).parent.joinpath("config.ini")
 
     def __init__(self, server_cfg: Path = None):
         self._server = (
-            Path(__file__).resolve().parent.parent.parent.joinpath("server.cfg")
+            Path().resolve().joinpath("server.cfg")
```

Comment:
I think I would rather have the `server.cfg` file in the top-level dir for better accessibility and configurability.
Or does this work even if that is the case?

Reply:
If you're running the server from the top-level directory anyway (which I think was implied in all our previous instructions) then this approach is equivalent.

What it means for me (which may not be suitable for everyone!) is that I can use the same `optimade.config` submodule in my implementation, without having to "patch" (in some sense) the optimade library files with my desired `server.cfg`. Of course, if you're not doing an editable install then the top-level optimade dir doesn't exist!

Reply:
Hmm. It seems we should have a more streamlined solution for this then? Since this PR is for making the package more usable as a library, then I wouldn't expect one to install it doing an editable install.
In my opinion it should work with both then, I guess? But maybe not? Maybe we insist on an editable install if one wants to use the package to run a server.

Reply:
I'm not sure what you mean, the result of this PR is that you should be able to do either an editable or non-editable install, and you can now use the `optimade.config` module from any directory. Ideally we'd have a way to specify the path to the `server.cfg` file, but obviously that's what this file is actually _for_!

Reply:
Right. Sorry. I think I'm just confused as to what path this line will produce in general. Will it produce the cwd + `/server.cfg` or will it produce cwd + `/optimade/server/server.cfg` or something different?

Reply:
So I've just tested it, and it does indeed run in cwd + `/server.cfg`, so if the file is at `/optimade/optimade-python-tools/server.cfg` and I am in `/optimade` running `./optimade-python-tools/run.sh` it will not work. I.e., I will always have to have the `server.cfg` in my cwd, irregardless of how I run `run.sh`.

I think that's actually better than what we had. But we _need_ to write this in the README.

Reply:
> So I've just tested it, and it does indeed run in cwd + `/server.cfg`, so if the file is at `/optimade/optimade-python-tools/server.cfg` and I am in `/optimade` running `./optimade-python-tools/run.sh` it will not work. I.e., I will always have to have the `server.cfg` in my cwd, irregardless of how I run `run.sh`.
> 
> I think that's actually better than what we had. But we _need_ to write this in the README.

I've just modified it so that `server_template.cfg` is not used if `server.cfg` is missing, as it was leading to incorrect relative paths in the generated config. Instead it uses the fallback values to write the new config file. I've left `server_template.cfg` in the `server/config` folder for instructional purposes, but I guess it could even be moved back to the top-level now its no longer needed as package data...

I think this is the most consistent way of doing this, I'll make a note in the README.

Reply:
Okay, so this tab only refreshed after writing this comment (and pushing the commit) and I now see that this PR has already been merged... I'll make another quick one @CasperWA!